### ISSUE TITLE
[FE] Add support for Dynamic Expiration date

### ISSUE THREAD

**Issue by mdewey** [department-of-veterans-affairs/va.gov-team#16938]
## User Story
As a veteran, 
I should have my questionnaire expire on the day of my appointment. 
So that I know when I can fill out the form by

## Related Documentation
 - [MVP: Visit Intro Initiative (Proof of Concept)](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/questionnaire/product/initiatives/visit-intro.md)
- [Mock up](https://app.zeplin.io/project/5f3d86a458ea2231e60a4780/screen/5f7b3ab94e155b94041d4c4b)
- [SPIKE Ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/15257)

## Description
The default form system has a 60 sliding expiration date for a form's data. For our use cases, 60 days is not correct. A questionnaire expires on the day of the appointment. This is about passing the expiration date to the SIP system. 

## Tasks
> this is a list of technical tasks that are needed for this ticket 
- [x] Add a hidden field to the form called `days_till_expires`, [proof of concept](https://github.com/department-of-veterans-affairs/vets-website/blob/healthcare-15257-sip-date/src/applications/healthcare/questionnaire/components/days-till-expires/DaysTillExpires.jsx)
- [x] ~[DELIGHTER]: create it in such a way to be a hidden form field and maybe contributed to the forms system.~ 
- [x] this should be auto-saved as well.  
- [x] Create user tests
- [x] Create e2e tests 

## Acceptance Criteria
- [x] Form as a dynamic expiration date
- [x] Passes Accessibility checks using axe
- [ ] PR has been merged into master
- [ ] Designer has signed off on the feature
- [ ] Product owner has signed off feature

**Comment by mdewey** [comment]
` [DELIGHTER]: create it in such a way to be a hidden form field and maybe contributed to the forms system.` Not possible until we update to the latest version of redux (v7.X) to use `useSelector` to query the store dynamically.

**Comment by mdewey** [comment]
Merged, waiting for staging deployment

**Comment by mdewey** [comment]
Blocked due to testing. Meeting with the engineering team to figure out what the next steps are.

**Comment by mdewey** [comment]
Back end ticket: https://github.com/department-of-veterans-affairs/va.gov-team/issues/17206

**Comment by mdewey** [comment]
TL;DR; on testing. The only changes were on the API but the QA team only has the capabilities to test the front end.
